pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -545,6 +545,38 @@ cdef class {{name}}HashTable(HashTable):
                     self.table.vals[k] = i
         self.na_position = na_position
 
+    @cython.boundscheck(False)
+    def hash_inner_join(self, const {{dtype}}_t[:] values, const uint8_t[:] mask = None) -> tuple[ndarray, ndarray]:
+        cdef:
+            Py_ssize_t i, n = len(values)
+            int ret = 0, ctr = 0
+            {{c_type}} val
+            khiter_t k
+            intp_t[::1] locs = np.empty(n, dtype=np.intp)
```

Comment:
The Vector classes would be a better fit than the allocate-in-full + trim strategy

Reply:
Oh, I didn't think about them. You are definitely correct, thx. Changed